client/X11/xf_rail.c

```diff
@@ -1232,12 +1231,13 @@ int xf_rail_init(xfContext* xfc, RailClientContext* rail)
 	rail->ServerMinMaxInfo = xf_rail_server_min_max_info;
 	rail->ServerLanguageBarInfo = xf_rail_server_language_bar_info;
 	rail->ServerGetAppIdResponse = xf_rail_server_get_appid_response;
-	xfc->railWindows = HashTable_New(TRUE);
 
+	xfc->railWindows = HashTable_New(TRUE);
 	if (!xfc->railWindows)
 		return 0;
-
+	xfc->railWindows->keyCompare = (HASH_TABLE_KEY_COMPARE_FN)rail_window_key_equals;
```

Comment:
Better not cast function pointers, that silences the compiler when the signature changes. Better do the casts in the implementation

Reply:
+1 for not casting. In case we have changes on the API the compiler wouldn't complain.

Reply:
I will do the change, but who would be mad enough to change the API. Well it could be me to change the equals_key function to take `const void *` in argument instead of `void *`.

Reply:
@hardening I had the same thought ;)